numba_dpex/core/runtime/_dpexrt_python.c

```diff
@@ -1269,6 +1273,97 @@ static PyObject *DPEXRT_sycl_queue_to_python(queuestruct_t *queuestruct)
     return orig_queue;
 }
 
+/*----------------------------------------------------------------------------*/
+/*--------------------- Box-unbox helpers for dpctl.SyclEvent       ----------*/
+/*----------------------------------------------------------------------------*/
+
+/*!
+ * @brief Helper to unbox a Python dpctl.SyclEvent object to a Numba-native
+ * eventstruct_t instance.
+ *
+ * @param    obj            A dpctl.SyclEvent Python object
+ * @param    event_struct   An instance of the struct numba-dpex uses to
+ *                          represent a dpctl.SyclEvent inside Numba.
+ * @return   {return}       Return code indicating success (0) or failure (-1).
+ */
+static int DPEXRT_sycl_event_from_python(PyObject *obj,
+                                         eventstruct_t *event_struct)
+{
+
+    struct PySyclEventObject *event_obj = NULL;
+    DPCTLSyclEventRef event_ref = NULL;
+    PyGILState_STATE gstate;
+
+    // We are unconditionally casting obj to a struct PySyclEventObject*. If
+    // the obj is not a struct PySyclEventObject* then the SyclEvent_GetEventRef
+    // will error out.
+    event_obj = (struct PySyclEventObject *)obj;
+
+    DPEXRT_DEBUG(
+        drt_debug_print("DPEXRT-DEBUG: In DPEXRT_sycl_event_from_python.\n"););
+
+    if (!(event_ref = SyclEvent_GetEventRef(event_obj))) {
+        DPEXRT_DEBUG(drt_debug_print(
+            "DPEXRT-ERROR: SyclEvent_GetEventRef returned NULL at "
+            "%s, line %d.\n",
+            __FILE__, __LINE__));
+        goto error;
+    }
+
+    event_struct->parent = obj;
+    event_struct->event_ref = event_ref;
+
+    return 0;
+
+error:
+    // If the check failed then decrement the refcount and return an error
+    // code of -1.
+    // Decref the Pyobject of the array
```

Comment:
The comment needs to be removed as well.

Reply:
Let's reconsider if we still need this inc/dec ref. Still the case when it used asynchronously from the python flow may cause gc to deallocate object. It is not a case with numpy in numba because we are doing incref for memory data.

Reply:
We need to check if numba does it for us out of the box.